### PR TITLE
Improve passlib CryptContext typing

### PR DESCRIPTION
**Pull request by hasier**
Reference: https://foss.heptapod.net/python-libs/passlib/-/blob/01dfe753e4f4590307611ef5c6304279a7364305/passlib/context.py

### CODE REVIEW

File: stubs/passlib/passlib/context.pyi [hunk 1]
```diff
@@ -25,45 +25,53 @@ class CryptPolicy:
...
     def replace(self, **kwds): ...
     def __init__(self, schemes: Any | None = ..., policy=..., _autoload: bool = ..., **kwds) -> None: ...
-    policy: Any
-    def load_path(self, path, update: bool = ..., section: str = ..., encoding: str = ...): ...
-    def load(self, source, update: bool = ..., section: str = ..., encoding: str = ...) -> None: ...
-    def update(self, *args, **kwds) -> None: ...
+    policy: CryptPolicy
+    def load_path(self, path: str, update: bool = ..., section: str = ..., encoding: str = ...) -> None: ...
+    def load(
+        self, source: str | bytes | dict[str, Any] | CryptContext, update: bool = ..., section: str = ..., encoding: str = ...
```
Comment (srittau) [inline stubs/passlib/passlib/context.pyi:38]: Instead of `dict`, we can use `_typeshed.SupportsItems` here.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks! Notes below.

File: stubs/passlib/passlib/context.pyi [hunk 2]
```diff
@@ -25,45 +25,53 @@ class CryptPolicy:
...
+    def copy(self, **kwds: Any) -> CryptContext: ...
+    def using(self, **kwds: Any) -> CryptContext: ...
     def replace(self, **kwds): ...
     def __init__(self, schemes: Any | None = ..., policy=..., _autoload: bool = ..., **kwds) -> None: ...
-    policy: Any
-    def load_path(self, path, update: bool = ..., section: str = ..., encoding: str = ...): ...
-    def load(self, source, update: bool = ..., section: str = ..., encoding: str = ...) -> None: ...
-    def update(self, *args, **kwds) -> None: ...
+    policy: CryptPolicy
+    def load_path(self, path: str, update: bool = ..., section: str = ..., encoding: str = ...) -> None: ...
```
[the remark this answers is not in the document]
Comment (hasier) [reply stubs/passlib/passlib/context.pyi:36]: Why the `| int` here?

File: stubs/passlib/passlib/context.pyi [hunk 3]
```diff
@@ -25,45 +25,53 @@ class CryptPolicy:
 
 class CryptContext:
     @classmethod
-    def from_string(cls, source, section: str = ..., encoding: str = ...): ...
+    def from_string(cls, source: str | bytes, section: str = ..., encoding: str = ...) -> CryptContext: ...
```
[the remark this answers is not in the document]
Comment (hasier) [reply stubs/passlib/passlib/context.pyi]: Done https://github.com/python/typeshed/pull/8237/commits/d70e371641f963b9606d9928f70cd16ff97b01b7

File: stubs/passlib/passlib/context.pyi [hunk 4]
```diff
@@ -25,45 +25,53 @@ class CryptPolicy:
...
     @classmethod
-    def from_path(cls, path, section: str = ..., encoding: str = ...): ...
-    def copy(self, **kwds): ...
-    def using(self, **kwds): ...
+    def from_path(cls, path: str, section: str = ..., encoding: str = ...) -> CryptContext: ...
```
[the remark this answers is not in the document]
Comment (hasier) [reply stubs/passlib/passlib/context.pyi]: Done https://github.com/python/typeshed/pull/8237/commits/d70e371641f963b9606d9928f70cd16ff97b01b7

File: stubs/passlib/passlib/context.pyi (hunk 2 above)
[the remark this answers is not in the document]
Comment (hasier) [reply stubs/passlib/passlib/context.pyi]: Done https://github.com/python/typeshed/pull/8237/commits/d70e371641f963b9606d9928f70cd16ff97b01b7 except the `int`, not too sure what it's for 😅

File: stubs/passlib/passlib/context.pyi (hunk 1 above)
Comment (hasier) [reply stubs/passlib/passlib/context.pyi]: Done https://github.com/python/typeshed/pull/8237/commits/d70e371641f963b9606d9928f70cd16ff97b01b7

File: stubs/passlib/passlib/context.pyi [hunk 5]
```diff
@@ -25,17 +26,17 @@ class CryptPolicy:
...
     def copy(self, **kwds: Any) -> CryptContext: ...
     def using(self, **kwds: Any) -> CryptContext: ...
     def replace(self, **kwds): ...
     def __init__(self, schemes: Any | None = ..., policy=..., _autoload: bool = ..., **kwds) -> None: ...
     policy: CryptPolicy
-    def load_path(self, path: str, update: bool = ..., section: str = ..., encoding: str = ...) -> None: ...
+    def load_path(self, path: StrOrBytesPath, update: bool = ..., section: str = ..., encoding: str = ...) -> None: ...
     def load(
-        self, source: str | bytes | dict[str, Any] | CryptContext, update: bool = ..., section: str = ..., encoding: str = ...
+        self, source: str | bytes | SupportsItems | CryptContext, update: bool = ..., section: str = ..., encoding: str = ...
```
Comment (srittau) [inline stubs/passlib/passlib/context.pyi]:
```suggestion
        self, source: str | bytes | SupportsItems[str, Any] | CryptContext, update: bool = ..., section: str = ..., encoding: str = ...
```

**Review by srittau — CHANGES_REQUESTED** [review]
One fix, LGTM otherwise.

File: stubs/passlib/passlib/context.pyi (hunk 2 above)
[the remark this answers is not in the document]
Comment (srittau) [reply stubs/passlib/passlib/context.pyi]: You can pass an FD to `open()`, not really important, though. But we should probably add a typealias to `_typeshed` for that.

**Review by srittau — APPROVED** [review]

**Pull request merged**